### ISSUE TITLE
Swift package manager isn't updated for Swift 4.2 - FIX INCLUDED IN COMMENTS

### ISSUE THREAD

**Issue by hmlongco** [ReactiveX/RxSwift#1677]
Doing a PR, but running tests locally goes into loop.

`Package for Swift package manager isn't updated, please run ./scripts/package-spm.swift and commit the changes`

I do so, commit, run it again and get the same error.

https://github.com/ReactiveX/RxSwift/pull/1676

Running tests in Xcode succeeds, but PR is failing with same error on Travis.

**Comment by hmlongco** [comment]
This is probably the cause...

Swift 4.2 implements SE-0206, which changes the way they make custom types conform to the Hashable protocol. It introduces a new Hasher struct that provides a randomly seeded, universal hash function to make this process easier:

Hasher uses a random seed every time it hashes an object, which means the hash value for any object is effectively guaranteed to be different between runs of your app.

**This in turn means that elements you add to a set or a dictionary are highly likely to have a different order each time you run your app.**

Thus I suspect that due to this, the function buildAllTestsTarget in ./scripts/package-spm.swift is no longer deterministic, and as such generates a different result every time that it's run.

**Comment by hmlongco** [comment]
Due to the hash changes, buildAllTestsTarget in ./scripts/package-spm.swift is no longer deterministic, and as such generates a different result every time that it's run.

**package-spm.swift needs the following changes:**

```
-    for (name, methods) in reducedMethods {

+    for name in reducedMethods.keys.sorted() {
+        let methods = reducedMethods[name]!
```

and

```
-    for testCase in reducedMethods.keys {

+    for testCase in reducedMethods.keys.sorted() {
```

**Comment by hmlongco** [comment]
PR #1676 `Add bag to dispose bag` includes this fix. After adding all checks passed.

**Comment by hmlongco** [comment]
PR up with this fix: #1678
